src/workflows-app/SubmissionDetails.test.js

```diff
@@ -298,45 +196,45 @@ describe('Submission Details page', () => {
 
     // Assert
     const table = await screen.findByRole('table');
-    const rows = within(table).queryAllByRole('row');
+    const rows = within(table).getAllByRole('row');
     expect(rows.length).toBe(3);
 
-    const headers = within(rows[0]).queryAllByRole('columnheader');
+    const headers = within(rows[0]).getAllByRole('columnheader');
     expect(headers.length).toBe(3);
 
     // Act - click on sort button on Duration column to sort by ascending order
-    user.click(await within(headers[2]).findByRole('button'));
+    await user.click(await within(headers[2]).findByRole('button'));
 
     // Assert
     // check that rows are now sorted by duration in ascending order
-    const cellsFromDataRow1 = within(rows[1]).queryAllByRole('cell');
+    const cellsFromDataRow1 = within(rows[1]).getAllByRole('cell');
     expect(cellsFromDataRow1.length).toBe(3);
     await within(cellsFromDataRow1[0]).findByText('FOO1');
```

Comment:
Should `FOO1` already be rendered at this point? So no need to `await` and `findBy`?